### ISSUE TITLE
wrong output from scipy.spatial.SphericalVoronoi

### ISSUE THREAD

**Issue by mortengryning** [scipy/scipy#11106]
The output from the spatial.SpheicalVoronoi algorithm seems wrong for some inputs. For example. consider the below input that are points on a sphere edge.

#### Reproducing code example:

```
...
points = np.array([
[297.2422124681002, 808.7477577269484, 80.30007256674854],    
[345.0781318294937, 732.8346696246801, -254.3053897848541], 
[708.4866213686736, 687.2893475336546, -253.6533638192258], 
[250.31425358851862,  216.3544010573513, 9.284511805790428], 
[632.9639320938492, 767.5530794227693, 231.55980305135643], 
])

radius = 378
center = np.array([500, 500, 0])
sv = SphericalVoronoi(points, radius, center)
...
```

#### Error:

Python plot of the voronoi regions
![image](https://user-images.githubusercontent.com/7712296/69426594-e0893800-0d2d-11ea-8efa-d903875138d2.png)

The regions mapped on a sphere (red lines are computed great circles to better show the regions) shown from 2 different perspectives:
![image](https://user-images.githubusercontent.com/7712296/69426865-8b015b00-0d2e-11ea-8064-a89b6b5e5dad.png)

![image](https://user-images.githubusercontent.com/7712296/69427005-d451aa80-0d2e-11ea-9489-7a139cac16fc.png)

#### Scipy/Numpy/Python version information:
```
1.2.0 1.16.1 sys.version_info(major=3, minor=6, micro=6, releaselevel='final', serial=0)
```

**Comment by tylerjereddy** [comment]
I'll try to check this later today---if you have a chance you may have better luck with the improved handling of `SphericalVoronoi` in a more recent version of SciPy; i.e., `python -m pip install scipy==1.4.0rc1` for our current release candidate.

Some of the improvements from @pmla there including handling single-hemisphere generators more gracefully---not sure if that is the exact issue here until I take a closer look.

**Comment by tylerjereddy** [comment]
It is perhaps less obvious there is an issue when using a recent version of SciPy; I normalized & centered your generators first. There could still be something slightly off, but not immediately obvious? I used the feature branch from #10380 to Slerp the region curves.

![test](https://user-images.githubusercontent.com/7903078/69599145-3af1f500-0fc9-11ea-8acd-54075d788abb.png)

There are some known limitations---input similar to the Tropic of Capricorn, I think.
